osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -160,20 +160,24 @@ private void applyDefaults(HitObject hitObject)
 
         private void applySamples(HitObject hitObject)
         {
-            SampleControlPoint sampleControlPoint = (beatmap.ControlPointInfo as LegacyControlPointInfo)?.SamplePointAt(hitObject.GetEndTime() + control_point_leniency) ?? SampleControlPoint.DEFAULT;
-
-            hitObject.Samples = hitObject.Samples.Select(o => sampleControlPoint.ApplyTo(o)).ToList();
-
             if (hitObject is IHasRepeats hasRepeats)
             {
+                SampleControlPoint sampleControlPoint = (beatmap.ControlPointInfo as LegacyControlPointInfo)?.SamplePointAt(hitObject.StartTime + CONTROL_POINT_LENIENCY + 1) ?? SampleControlPoint.DEFAULT;
```

Comment:
What's the significance of the +1 here, and it not being added in the other path?

Reply:
The zero-th node of the slider, i.e. the head, queries the control point at start time + 5ms (that is the leniency that is carried over from stable). That's where this extra +1 comes from; it's to ensure that the two don't get confused.